src/gui/invalidfilenamedialog.cpp

```diff
@@ -0,0 +1,169 @@
+
+    connect(_ui->filenameLineEdit, &QLineEdit::textChanged, this,
+        &InvalidFilenameDialog::onFilenameLineEditTextChanged);
+
+    checkIfAllowedToRename();
+}
+
+InvalidFilenameDialog::~InvalidFilenameDialog() = default;
+
+void InvalidFilenameDialog::checkIfAllowedToRename()
+{
+    const auto propfindJob = new PropfindJob(_account, QDir::cleanPath(_folder->remotePath() + _originalFileName));
+    propfindJob->setProperties({ "http://owncloud.org/ns:permissions" });
+    connect(propfindJob, &PropfindJob::result, this, &InvalidFilenameDialog::onPropfindPermissionSuccess);
+    propfindJob->start();
+}
+
+void InvalidFilenameDialog::onPropfindPermissionSuccess(const QVariantMap &values)
+{
+    if (!values.contains("permissions")) {
+        return;
+    }
+    const auto remotePermissions = RemotePermissions::fromServerString(values["permissions"].toString());
+    if (!remotePermissions.hasPermission(remotePermissions.CanRename)
+        || !remotePermissions.hasPermission(remotePermissions.CanMove)) {
+        _ui->errorLabel->setText(
+            tr("You don't have the permission to rename this file. Please ask the author of the file to rename it."));
+        _ui->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(false);
+        _ui->filenameLineEdit->setEnabled(false);
+    }
+}
+
+void InvalidFilenameDialog::accept()
+{
+    _newFilename = _relativeFilePath + _ui->filenameLineEdit->text().trimmed();
+    const auto propfindJob = new PropfindJob(_account, QDir::cleanPath(_folder->remotePath() + _newFilename));
+    connect(propfindJob, &PropfindJob::result, this, &InvalidFilenameDialog::onPropfindSuccess);
+    connect(propfindJob, &PropfindJob::finishedWithError, this, &InvalidFilenameDialog::onPropfindError);
+    propfindJob->start();
+}
+
+void InvalidFilenameDialog::onFilenameLineEditTextChanged(const QString &text)
+{
+    const auto isNewFileNameDifferent = text != _originalFileName;
+    const auto illegalContainedCharacters = getIllegalCharsFromString(text);
+    const auto containsIllegalChars = !illegalContainedCharacters.empty() || text.endsWith(QLatin1Char('.'));
+    const auto isTextValid = isNewFileNameDifferent && !containsIllegalChars;
+
+    if (isTextValid) {
+        _ui->errorLabel->setText("");
+    } else {
+        _ui->errorLabel->setText(tr("Filename contains illegal characters: %1")
+                                     .arg(illegalCharacterListToString(illegalContainedCharacters)));
+    }
+
+    _ui->buttonBox->button(QDialogButtonBox::Ok)
+        ->setEnabled(isTextValid);
+}
+
+void InvalidFilenameDialog::onMoveJobFinished()
+{
+    const auto job = qobject_cast<MoveJob *>(sender());
+    const auto error = job->reply()->error();
+
+    if (error != QNetworkReply::NoError) {
+        _ui->errorLabel->setText(tr("Could not rename file. Please make sure you are connected to the server."));
+        return;
+    }
+
+    QDialog::accept();
```

Comment:
why do you not trigger a sync run ?
from my understanding, you are changing the name on the  server but that rename also need to be locally synced ?

Reply:
The sync is triggered after the dialog finished. Not sure why I did it that way, but I did it mainly that way because the conflict dialog does it the same way.

Reply:
@FlexW which part of the code does that ?
how do you make sure that the rename is properly detected and that we cannot end up with the file being duplicated ?

Reply:
I added a comment where it is done. First I check if the file already exists on the server or locally. If it does not exist locally or on the server, a rename job is sent to the server. The server then renames the file and then new sync gets triggered.